examples/virtual_methods/base_button/imp.rs

```diff
@@ -0,0 +1,115 @@
+use super::BaseButtonExt;
+use gtk::{
+    gio,
+    glib::{self, translate::from_glib_borrow, Error},
+    prelude::*,
+    subclass::prelude::*,
+};
+use std::{future::Future, pin::Pin};
+
+pub type BaseButtonInstance = <BaseButton as super::ObjectSubclass>::Instance;
+pub type PinnedFuture = Pin<Box<dyn Future<Output = Result<(), Error>> + 'static>>;
+
+/// GObject glue code for our BaseButtonClass which holds the function pointers to our virtual functions.
+#[repr(C)]
+pub struct BaseButtonClass {
+    pub parent_class: gtk::ffi::GtkButtonClass,
+    pub sync_method:
+        Option<unsafe extern "C" fn(*mut BaseButtonInstance, extra_text: *mut Option<String>)>,
```

Comment:
If you wanted, you could express the `None` by `ptr::null_mut()` and then only have `*mut String` here

Reply:
I personally think it's nice using Option<T> here so it's clear that this can be None.

Reply:
Works for me